tests/e2e/statefulset_xfs.go

```diff
@@ -0,0 +1,218 @@
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	fpod "k8s.io/kubernetes/test/e2e/framework/pod"
+	fss "k8s.io/kubernetes/test/e2e/framework/statefulset"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+/*
+	Test performs following operations
+
+	Steps
+	1. Create a storage class with fstype set to "xfs".
+	2. Create nginx service.
+	3. Create nginx statefulset with 3 replicas.
+	4. Wait until all Pods are ready and PVCs are bounded with PV.
+	5. Scale down statefulset to 2 replicas.
+	6. Scale up statefulset to 3 replicas.
+	7. Scale down statefulset to 0 replicas and delete all pods.
+	8. Delete all PVCs from the test's namespace.
+	9. Delete the storage class.
+*/
+
+var _ = ginkgo.Describe("[csi-block-vanilla] [csi-block-vanilla-parallelized] statefulset with XFS filesystem", func() {
+
+	f := framework.NewDefaultFramework("e2e-vsphere-statefulset")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+	var (
+		namespace        string
+		client           clientset.Interface
+		scParameters     map[string]string
+		storageClassName string
+	)
+	ginkgo.BeforeEach(func() {
+		namespace = getNamespaceToRunTests(f)
+		client = f.ClientSet
+		bootstrap()
+		scParameters = make(map[string]string)
+	})
+
+	ginkgo.AfterEach(func() {
+		ginkgo.By(fmt.Sprintf("Deleting all statefulsets in namespace: %v", namespace))
+		fss.DeleteAllStatefulSets(client, namespace)
+	})
+
+	ginkgo.It("Statefulset testing with XFS filesystem", func() {
+		ginkgo.By("CNS_TEST: Running for vanilla k8s setup")
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By("Creating StorageClass for Statefulset with fstype set to xfs")
+		scParameters[scParamFsType] = xfsFSType
+		storageClassName = "nginx-sc-default"
+		scSpec := getVSphereStorageClassSpec(storageClassName, scParameters, nil, "", "", false)
+		sc, err := client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Creating service")
+		service := CreateService(namespace, client)
+		defer func() {
+			deleteService(namespace, client, service)
+		}()
+
+		ginkgo.By("Creating statefulset")
+		statefulset := GetStatefulSetFromManifest(namespace)
+		statefulset.Spec.VolumeClaimTemplates[len(statefulset.Spec.VolumeClaimTemplates)-1].
+			Annotations["volume.beta.kubernetes.io/storage-class"] = storageClassName
+		CreateStatefulSet(namespace, statefulset, client)
+		replicas := *(statefulset.Spec.Replicas)
+		// Waiting for pods status to be Ready
+		fss.WaitForStatusReadyReplicas(client, statefulset, replicas)
+		gomega.Expect(fss.CheckMount(client, statefulset, mountPath)).NotTo(gomega.HaveOccurred())
+		ssPodsBeforeScaleDown := fss.GetPodList(client, statefulset)
```

Comment:
can we check  and confirm the FS type on the volumes here

Reply:
done